File: 2021/0035_searchInsert/0809/searchInsert.ts
function searchInsert(nums: number[], target: number): number {
    let left = 0;
    let right = nums.length - 1;
    while (left <= right) {
        const mid = left + ((right - left) >> 1);
        if (nums[mid] == target) { return mid }
        if (nums[mid] > target) { right = mid - 1 }
        else { left = mid + 1}
    }
    return left;
}

File: 2021/0035_searchInsert/0809/searchInsert.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # use lib method
        # bisect.bisect_left(nums, target)
        l, r = 0, len(nums) - 1
        while l <= r:
            m = l + ((r - l) >> 1)
            if nums[m] == target:
                return m
            if nums[m] > target:
                r = m - 1
            else:
                l = m + 1
        return l
